unit_tests/test_prometheus.py

```diff
@@ -0,0 +1,154 @@
+        }
+        super().start()
+
+    def _post_alert(self, alert):
+        response = requests.post(self._alerts_endpoint, json=[alert], timeout=3)
+        if response.status_code != 200:
+            raise RuntimeError(f"Can't post alerts to '{self._alert_manager_url}/alerts' "
+                               f"status={response.status_code} response={response.content}")
+        return True
+
+    def post_next_alert(self):
+        self._current_alert_to_post += 1
+        if self._current_alert_to_post < len(self._artificial_alerts):
+            return self._post_alert(self._artificial_alerts[self._current_alert_to_post])
+        if self._current_alert_to_post > len(self._artificial_alerts) * 2:
+            self._stop_flag.set()
+            return False
+        self._stop_flag.set()
+        return False
+
+    def wait_alert_event(self, event_type: str, timeout=None):
+        if timeout is None:
+            timeout = self._events_wait_timeout
+        end_time = time.time() + timeout
+        current_alert = PrometheusAlertManagerEvent(event_type=event_type,
+                                                    raw_alert=self._artificial_alerts[self._current_alert_to_post])
+        while True:
+            if time.time() > end_time:
+                raise TimeoutError(f"Waiting for alert event #{self._current_alert_to_post} reached timeout: {str(current_alert)}")
+            line = self._event_log_file.readline()
+            if not line:
+                continue
+            i = PrometheusAlertManagerEvent()
+            if not i._from_event_str(line):
+                continue
+            if current_alert == i:
+                return True
+
+    def post_alerts_and_read_events(self, timeout=60):
+        alert = self.post_next_alert()
+        end_time = time.time() + timeout
+        while alert is not False:
+            self.wait_alert_event(event_type='start')
+            alert = self.post_next_alert()
+            if time.time() > end_time:
+                raise TimeoutError("post_alerts_and_read_events reached timeout")
+
+    def stop(self, timeout=None):
+        super().stop(timeout)
+        stop_events_device()
+
+
+class TestSeedSelector(unittest.TestCase):
+    temp_dir = None
+
+    @classmethod
+    def setUpClass(cls):
+        cls.temp_dir = tempfile.mkdtemp()
+
+    @classmethod
+    def tearDownClass(cls):
+        # stop_events_device()
+        shutil.rmtree(cls.temp_dir)
+
+    def test_alert_manager_listener_artificial_run(self):
+        with open('test_data/test_prometheus/test_alert_manager_listener_artificial_run.yaml') as f:
+            test_data = json.load(f)
+        i = PrometheusAlertManagerListenerArtificialTest(artificial_alerts=test_data['post'])
+        i.start()
+        result = i.get_result()
+        self.assertEqual(result, test_data['expected'])
+
+    @unittest.skip("TBD: dockerized prometheuse to be added")
+    def test_alert_manager_listener_real_run(self):
```

Comment:
:+1:  for manual kind of test, they really help next time we need to debug this part